boxoffice/views/admin_discount.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+import math
+from flask import jsonify, request
+from .. import app, lastuser
+from coaster.views import load_models, render_with, requestargs
+from coaster.utils import buid
+from ..models import db
+from boxoffice.models import Organization, ItemCollection, DiscountPolicy, DiscountCoupon, Item, Price, CURRENCY_SYMBOL
+from ..forms import DiscountPolicyForm, DiscountCouponForm
+from utils import xhr_only, date_time_format, api_error, api_success
+
+
+def jsonify_price(price):
+    if price:
+        return {
+            'price_title': price.title,
+            'amount': price.amount,
+            'start_at': date_time_format(price.start_at),
+            'end_at': date_time_format(price.end_at)
+        }
+    else:
+        return None
+
+
+def jsonify_discount_policy(policy):
+    return {
+        'id': policy.id,
+        'name': policy.name,
+        'title': policy.title,
+        'discount_type': "Automatic" if policy.is_automatic else "Coupon based",
+        'item_quantity_min': policy.item_quantity_min,
+        'percentage': policy.percentage,
+        'is_price_based': policy.is_price_based,
+        'discount_code_base': policy.discount_code_base,
+        'bulk_coupon_usage_limit': policy.bulk_coupon_usage_limit,
+        'discount': policy.percentage if not policy.is_price_based else '',
+        'price_details': jsonify_price(Price.query.filter(Price.discount_policy == policy).first()) if policy.is_price_based else '',
+        'currency': CURRENCY_SYMBOL['INR'],
+        'dp_items': [{'id': str(item.id), 'title': item.title} for item in policy.items]
```

Comment:
Probably won't make a significant difference, but generally, it's better to only load the attributes that you are going to use, which in this case, are `id` and `title`.